src/extensions/wcf/Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater/UpdaterFactory.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.WCFUpdater
+{
+    public class UpdaterFactory
+    {
+        public static PackageUpdater? GetPackageUpdater(string path, ILogger logger)
```

Comment:
Rather than having static methods, this class could be made available in dependency injeciton (similar to the upgrade step) and other classes could request it in their constructors. That would also allow this class to get an instance of ILoggerFactory in its ctor which could be used to create the loggers for all the various updaters.